shotover-proxy/src/transforms/cassandra/sink_cluster/mod.rs

```diff
@@ -265,46 +273,87 @@ impl CassandraSinkCluster {
         }
 
         let mut responses_future = FuturesOrdered::new();
+
         let mut responses_future_use = FuturesOrdered::new();
         let mut use_future_index_to_node_index = vec![];
+
+        let mut responses_future_prepare = FuturesOrdered::new();
+
         for mut message in messages {
             let (return_chan_tx, return_chan_rx) = oneshot::channel();
-            if self.local_nodes.is_empty()
+            if self.pool.nodes.is_empty()
                 || !self.init_handshake_complete
                 // system.local and system.peers must be routed to the same node otherwise the system.local node will be amongst the system.peers nodes and a node will be missing
                 // DDL statements and system.local must be routed through the same connection, so that schema_version changes appear immediately in system.local
                 || is_ddl_statement(&mut message)
                 || self.is_system_query(&mut message)
             {
-                self.init_handshake_connection.as_mut().unwrap()
+                self.init_handshake_connection
+                    .as_mut()
+                    .unwrap()
+                    .send(message, return_chan_tx)?;
             } else if is_use_statement(&mut message) {
                 // Adding the USE statement to the handshake ensures that any new connection
                 // created will have the correct keyspace setup.
                 self.connection_factory.set_use_message(message.clone());
 
                 // Send the USE statement to all open connections to ensure they are all in sync
-                for (node_index, node) in self.local_nodes.iter().enumerate() {
-                    if let Some(outbound) = &node.outbound {
+                for (node_index, node) in self.pool.nodes.iter().enumerate() {
+                    if let Some(connection) = &node.outbound {
                         let (return_chan_tx, return_chan_rx) = oneshot::channel();
-                        outbound.send(message.clone(), return_chan_tx)?;
+                        connection.send(message.clone(), return_chan_tx)?;
                         responses_future_use.push_back(return_chan_rx);
                         use_future_index_to_node_index.push(node_index);
                     }
                 }
 
                 // Send the USE statement to the handshake connection and use the response as shotovers response
-                self.init_handshake_connection.as_mut().unwrap()
+                self.init_handshake_connection
+                    .as_mut()
+                    .unwrap()
+                    .send(message, return_chan_tx)?;
+            } else if is_prepare_message(&mut message) {
+                // Send the PREPARE statement to all connections
+                for node in self.pool.nodes.iter_mut() {
+                    let (return_chan_tx, return_chan_rx) = oneshot::channel();
+
+                    node.get_connection(&self.connection_factory)
```

Comment:
This will open a connection against every node in the cluster.
I'm not sure if this approach or doing what we do with use statement handling is better where we only use a connection if it already exists.
The current approach is better when the prepared statement is unpopulated and the execute will hit most nodes eventually.
Not opening every connection is better in cases where the prepared statement was already fully populated by a previous session.

We can just leave it as is for now, and think about this more when we need to improve performance

Reply:
I think this is done to avoid the following behavior:
- Send prepared statement
- Not on the routed node
- return error to client
- client responds with prepare message
- shotover routes it to a different node

So as the size of the cluster increases, the number of errors shotover returns per unrecognized statement will increase by an average of (N - Nodes_already_prepped)/2 where N == number of nodes in the cluster.

So the tradeoff is, do we have an upfront hit in latency when we try to do this for every node in the cluster (with this approach it's also done sequentially rather than concurrently) or do bounce back and forth for every prepared query.

Reply:
But yes, lets leave this in and figure out a better way to do it later (at a minimum this needs to send out the messages to the nodes in the cluster concurrently)

Reply:
yep that make sense to me, lets keep this approach.
Your absolutely right about the connections being opened sequentially instead of concurrently?
@conorbros Do you want to make that concurrent here, or just add a TODO and get to it in a follow up?

Reply:
I'll add it to the follow up list, should be fine for now since it will only be slow for the first prepared statement, in subsequent ones the connections will already be there.